network/p2p/libp2pNode.go

```diff
@@ -43,6 +45,36 @@ const (
 // maximum number of attempts to be made to connect to a remote node for 1-1 direct communication
 const maxConnectAttempt = 3
 
+// LibP2PFactoryFunc is a factory function type for generating libp2p Node instances.
+type LibP2PFactoryFunc func(allowList []NodeAddress) (*Node, error)
```

Comment:
```suggestion
type LibP2PFactoryFunc func() (*Node, error)
```

Reply:
So, we can always start the node with an empty or nil list. The unittest are actually doing that and then when we call `network.SetIDs()` the allow lis is updated.
This will keep the factory simple.

Reply:
fixed in https://github.com/onflow/flow-go/pull/195/commits/eb77e9fcbb1a1ff957a10a3a0a89d49ab07c8ff1